packages/modules/order/src/utils/transform-order.ts

```diff
@@ -5,18 +5,32 @@ import {
   deduplicate,
   isDefined,
 } from "@medusajs/utils"
+import { Order, OrderClaim, OrderExchange, Return } from "@models"
 
 export function formatOrder(
   order,
   options: {
+    entity: any
     includeTotals?: boolean
   }
-): OrderTypes.OrderDTO | OrderTypes.OrderDTO[] {
+): Partial<OrderTypes.OrderDTO> | Partial<OrderTypes.OrderDTO>[] {
   const isArray = Array.isArray(order)
   const orders = [...(isArray ? order : [order])]
 
   orders.map((order) => {
-    order.items = order.items?.map((orderItem) => {
+    let mainOrder = order
+
+    const isRelatedEntity = options?.entity !== Order
+
+    if (isRelatedEntity) {
+      if (!order.order) {
+        return order
+      }
```

Comment:
question: What case are we covering here?

Reply:
Returns, Claims and Exchanges have the entity `order`. Just in case it is not present we don't try to format it.